### PR TITLE
Remove repeated _check_fname() calls when saving Raw

### LINKED ISSUE #10094: Raw.save logs too many `"Overwriting existing file."` infos
MWE: 

```python
# %%
import os
from pathlib import Path
import mne
import numpy as np


sample_dir = Path(mne.datasets.sample.data_path())
sample_fname = sample_dir / 'MEG' / 'sample' / 'sample_audvis_raw.fif'
raw = mne.io.read_raw_fif(sample_fname, preload=True)

fname = Path(os.path.expanduser("~")) / "Desktop" / "try-raw.fif.gz"
raw.save(fname)
raw.save(fname, overwrite=True)

```

Output after second `.save` call:

```
Overwriting existing file.
Overwriting existing file.
Overwriting existing file.
Writing /home/stefanappelhoff/Desktop/try-raw.fif.gz
Closing /home/stefanappelhoff/Desktop/try-raw.fif.gz
[done]
```

These are two `Overwriting existing file.` too many, aren't they?

### PR DESCRIPTION
**Pull request by hoechenberger**
Fixes #10094

cc @sappelhoff

### CODE REVIEW

File: mne/io/base.py [hunk 1]
```diff
@@ -1426,9 +1427,19 @@ def save(self, fname, picks=None, tmin=0, tmax=None, buffer_size_sec=None,
...
+        # overwriting
+        fname = str(Path(fname).expanduser())
+
+        # check naming convention
+        check_fname(
+            fname=fname,
+            filetype='raw',
+            endings=endings,
+            endings_err=endings_err
+        )
```
Comment (larsoner) [inline mne/io/base.py:1442]: This code is less DRY, but also less functional as it won't _validate_type as path-like (right)? Instead can you just do:
`_check_fname(fname, overwrite=True, verbose='error')`? This won't log anything

Comment (hoechenberger) [reply mne/io/base.py:1442]: `_check_fname` currently doesn't have a `verbose` param, but could make sense to add one.

@sappelhoff, wanna take over? ;) I'll be busy with other things until tomorrow or so

File: mne/utils/check.py [hunk 2]
```diff
@@ -147,8 +147,9 @@ def _check_event_id(event_id, events):
     return event_id
 
 
+@verbose
 def _check_fname(fname, overwrite=False, must_exist=False, name='File',
-                 need_dir=False):
+                 need_dir=False, *, verbose=True):
```
Comment (larsoner) [inline mne/utils/check.py:152]: In general all new `verbose` calls should be
```suggestion
                 need_dir=False, *, verbose=None):
```
Having `verbose=True` here will probably break some `verbose=False` / `verbose='error'` usages elsewhere in our code

**Review by larsoner — APPROVED** [review]

**Comment by larsoner** [comment]
Thanks @sappelhoff !

**Comment by larsoner** [comment]
And @hoechenberger !

**Pull request merged**